plugins/rest/auth.go

```diff
@@ -145,22 +145,41 @@ func (ap *bearerAuthPlugin) NewClient(c Config) (*http.Client, error) {
 }
 
 func (ap *bearerAuthPlugin) Prepare(req *http.Request) error {
+	token, err := ap.buildToken()
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("%v %v", ap.Scheme, token))
+	return nil
+}
+
+func (ap *bearerAuthPlugin) AuthHeader() (http.Header, error) {
```

Comment:
Could you add a comment explaining what it's for?

Reply:
tried to do that, can you have another look

Reply:
Thank you! Looks good 👍